client-js/app/elements/labrad-registry.ts

```diff
@@ -42,15 +43,77 @@ export class LabradRegistry extends polymer.Base {
                                               this.$.editValueInput);
   }
 
+
+  getDefaultSelectedItem(): number {
+    return this.path.length > 0 ? 1 : 0;
+  }
+
+
+  cursorMove(e) {
+    const offset = this.path.length > 0 ? 1 : 0;
+    const length = this.keys.length + this.dirs.length + offset;
+
+    switch (e.detail.combo) {
+      case 'up':
+        if (this.selectedIdx === null || this.selectedIdx === 0) {
+          this.set('selectedIdx', null);
+        } else {
+          this.set('selectedIdx', this.selectedIdx - 1);
+        }
+        break;
+
+      case 'down':
+        if (this.selectedIdx === null) {
+          this.set('selectedIdx', 0);
+        } else if (this.selectedIdx < length - 1) {
+          this.set('selectedIdx', this.selectedIdx + 1);
+        }
+        break;
+
+      default:
+        // Nothing to do.
+        break;
+    }
+  }
+
+
+  cursorTraverse(e) {
+    if (this.selectedIdx === null) {
+      return;
+    }
+
+    const item = this.$.combinedList.selectedItem;
+    const link = item.querySelector('a');
+
+    // If we have a link, we want to traverse down.
+    if (link) {
+      this.fire('app-link-click', {path: link.href});
```

Comment:
This is causes a full page reload, rather than just doing a route change within the app. You want to use `link.path` instead.